rclcpp/include/rclcpp/memory_strategy.hpp

```diff
@@ -53,8 +53,33 @@ class MemoryStrategy
    */
   virtual void ** borrow_handles(HandleType type, size_t number_of_handles)
   {
-    (void)type;
-    return static_cast<void **>(alloc(sizeof(void *) * number_of_handles));
+    switch (type) {
+      case HandleType::subscription_handle:
+        if (subscription_handles.size() < number_of_handles) {
+          subscription_handles.resize(number_of_handles, 0);
+        }
+        return static_cast<void **>(subscription_handles.data());
+        break;
+      case HandleType::service_handle:
+        if (service_handles.size() < number_of_handles) {
+          service_handles.resize(number_of_handles, 0);
+        }
+        return static_cast<void **>(service_handles.data());
+        break;
+      case HandleType::client_handle:
+        if (client_handles.size() < number_of_handles) {
+          client_handles.resize(number_of_handles, 0);
+        }
+        return static_cast<void **>(client_handles.data());
+        break;
+      case HandleType::guard_condition_handle:
+        if (number_of_handles > 2) {
+          throw std::runtime_error("Too many guard condition handles requested!");
+        }
+        return guard_cond_handles.data();
+      default:
+        throw std::runtime_error("Unrecognized enum, could not borrow handle memory.");
```

Comment:
Since HandleType is an enum, how would you suggest I print its representation? As an integer?